cli/ops/tls.rs

```diff
@@ -196,18 +254,110 @@ fn op_listen_tls(
   config
     .set_single_cert(load_certs(&cert_file)?, load_keys(&key_file)?.remove(0))
     .expect("invalid key or certificate");
-  let acceptor = TlsAcceptor::from(Arc::new(config));
+  let tls_acceptor = TlsAcceptor::from(Arc::new(config));
   let addr = resolve_addr(&args.hostname, args.port).wait()?;
   let listener = TcpListener::bind(&addr)?;
   let local_addr = listener.local_addr()?;
-  let resource = resources::add_tls_listener(listener, acceptor);
+  let local_addr_str = local_addr.to_string();
+  let tls_listener_resource = TlsListenerResource {
+    listener,
+    tls_acceptor,
+    task: None,
+    local_addr,
+  };
+  let mut table = resources::lock_resource_table();
+  let rid = table.add("tlsListener", Box::new(tls_listener_resource));
 
   Ok(JsonOp::Sync(json!({
-    "rid": resource.rid,
-    "localAddr": local_addr.to_string()
+    "rid": rid,
+    "localAddr": local_addr_str
   })))
 }
 
+#[derive(Debug, PartialEq)]
+enum AcceptTlsState {
+  Eager,
```

Comment:
Given the fact that we do eager polling in core, I wonder if this can be removed?

Reply:
Good catch

Reply:
Oooops I forgot why it was there... I guess we have to keep it there. If we eager poll we shouldn't track task because it would be associated with that task. Then the future is added to pending futures (because it wasn't ready), it will be picked by by another task and panic that it's already tracked by other task.